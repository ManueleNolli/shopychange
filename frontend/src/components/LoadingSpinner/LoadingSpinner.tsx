import React from 'react'
import BeatLoader from 'react-spinners/BeatLoader'
import { Center, useColorMode } from '@chakra-ui/react'

export default function LoadingSpinner() {
  const { colorMode } = useColorMode()

  return (
    <Center mt="5vh">
      <BeatLoader size={8} color={colorMode === 'light' ? '#000' : '#fff'} />
    </Center>
  )
}
