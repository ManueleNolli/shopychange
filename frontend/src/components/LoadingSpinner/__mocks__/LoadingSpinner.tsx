import { divWithoutChildrenMock } from '../../../../mocksUtils'

const LoadingSpinnerMock = () => divWithoutChildrenMock('LoadingSpinner')
export default LoadingSpinnerMock
