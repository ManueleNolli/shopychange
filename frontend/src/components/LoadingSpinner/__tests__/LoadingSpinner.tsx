import React from 'react'
import { render } from '@testing-library/react'
import LoadingSpinner from '../LoadingSpinner'
import { useColorMode } from '@chakra-ui/react'

jest.mock('@chakra-ui/react', () => ({
  ...jest.requireActual('@chakra-ui/react'),
  useColorMode: jest.fn(),
}))

describe('LoadingSpinner', () => {
  test('renders LoadingSpinner correctly Light Mode', () => {
    ;(useColorMode as jest.Mock).mockImplementation(() => ({
      colorMode: 'light',
    }))

    const tree = render(<LoadingSpinner />)

    expect(tree.container).toMatchSnapshot()
  })
  test('renders LoadingSpinner correctly Dark Mode', () => {
    ;(useColorMode as jest.Mock).mockImplementation(() => ({
      colorMode: 'dark',
    }))
    const tree = render(<LoadingSpinner />)

    expect(tree.container).toMatchSnapshot()
  })
})
