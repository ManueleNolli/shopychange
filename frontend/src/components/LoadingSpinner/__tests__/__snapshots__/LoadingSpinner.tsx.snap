// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner renders LoadingSpinner correctly Dark Mode 1`] = `
<div>
  <div
    class="css-1ffdipv"
  >
    <span
      style="display: inherit;"
    >
      <span
        style="display: inline-block; background-color: rgb(255, 255, 255); width: 8px; height: 8px; margin: 2px; border-radius: 100%; animation: react-spinners-BeatLoader-beat 0.7s 0s infinite linear; animation-fill-mode: both;"
      />
      <span
        style="display: inline-block; background-color: rgb(255, 255, 255); width: 8px; height: 8px; margin: 2px; border-radius: 100%; animation: react-spinners-BeatLoader-beat 0.7s 0.35s infinite linear; animation-fill-mode: both;"
      />
      <span
        style="display: inline-block; background-color: rgb(255, 255, 255); width: 8px; height: 8px; margin: 2px; border-radius: 100%; animation: react-spinners-BeatLoader-beat 0.7s 0s infinite linear; animation-fill-mode: both;"
      />
    </span>
  </div>
</div>
`;

exports[`LoadingSpinner renders LoadingSpinner correctly Light Mode 1`] = `
<div>
  <div
    class="css-1ffdipv"
  >
    <span
      style="display: inherit;"
    >
      <span
        style="display: inline-block; background-color: rgb(0, 0, 0); width: 8px; height: 8px; margin: 2px; border-radius: 100%; animation: react-spinners-BeatLoader-beat 0.7s 0s infinite linear; animation-fill-mode: both;"
      />
      <span
        style="display: inline-block; background-color: rgb(0, 0, 0); width: 8px; height: 8px; margin: 2px; border-radius: 100%; animation: react-spinners-BeatLoader-beat 0.7s 0.35s infinite linear; animation-fill-mode: both;"
      />
      <span
        style="display: inline-block; background-color: rgb(0, 0, 0); width: 8px; height: 8px; margin: 2px; border-radius: 100%; animation: react-spinners-BeatLoader-beat 0.7s 0s infinite linear; animation-fill-mode: both;"
      />
    </span>
  </div>
</div>
`;
